# This notebook is for BCB430 
## Sep
### 03
“Vlasschaert et al. A practical approach to curate clonal hematopoiesis of indeterminate potential in human genetic data sets”
- variants: 74 CHIP genes
- filtering: depth >=20, minAD>=3, F1R2, F2R1>=1, VAF>=2%
- TERT and age removal
- binomial neal 50% removal
- minAD>=5 for UKB, >=3 for AOU
totally remove 16 CHIP genes

### 04
“Analysis of 10,478 cancer genomes identifies candidate driver genes and opportunities for precision oncology“
- dN/dS----driver genes
- COSMIC, intOGen TCGA databases test driver genes
- Clinical Actionability/Tumor-agnostic Biomarkers/Druggable Genome

### 09
2019 Science Review: *Clonal Hematopoiesis in Human Aging and Disease*  
1. Hematopoietic stem cells acquire **driver** mutations → CHIP, VAF ≥ 2%.  
2. Elevated risk of hematologic cancers **(possibly due to mutations in cancer-related genes?)**g g
3. Cardiovascular disease **(why?)**

2022 Nature Study: UK Biobank GWAS of CHIP  
1. GWAS and ExWAS.  
2. Not a single phenotype—multiple heterogeneous subtypes across different CHIP mutations.  
3. Different mutation types show distinct behavior; for example, **DNMT3A-type** CHIP tends to progress to myeloid leukemias such as AML.

### 10
2023 Clinical Cancer Research Review: Relationship between CHIP and Solid Tumors  
1. CHIP is restricted to asymptomatic individuals without a hematologic diagnosis.  
2. CHIP → clonal cytopenia of undetermined significance (CCUS) → overt malignancy.

### 11
1. How does paired-end short-read sequencing work?What are its strengths and limitations 
- sequencing from both end
- more sensitive
- expensive
2. Mutation types and how they can affect a protein (missense, loss of function etc.)
- missense: differen protein
- frameshift: loss of function
- nonsense: stop early
- silent: no effect
3. Germline vs somatic variants
- Germline: froma parent, 50%
- somaric: not be inherited
4. **What is clonal hematopoiesis?Why does it happen?**
- some driver mutation brings Proliferation advantage
- stem cell has mutation to grow in larger number
- mutated stem cell produce more mutated stem cell 
5. What is clonal hematopoiesis of indeterminate potential?How does it differ from regular clonal hematopoiesis 
- without diagonosis

6. What are cancer driver mutations?How do we find them? 
- mutation make cell have Proliferation advantage
- high frequency in cancer; prediction protein function; pathway analyze (mutual exclusivity); evolution?
7. What is a passenger mutation?
- no effect, random mutation
- uniform distribution
8. The genome sequencing bioinformatics workflow – major steps 
- FASTAQ--QC---alignment with ref---bim---variant calling---vcf---annotation
9. What are the factors affecting variant call quality?
- coverage, 
11. Differences in calling germline vs. somatic variants (bioinformatics)

### 21
1. Back to mutation types: what do you think are the most damaging mutation types? Least damaging? Rank the mutation types from before in order of how damaging they would be to a protein.
- insertion/deletion(frameshift) > nonsense > missense > slient
2. When we are trying to find mutations that are pathogenic, we often will look for mutations that are rare in a population database (like gnomAD) – why do we do this? Why is a rare mutation much more likely to be disease-causing than one that is very common?
- Most people don't have disease. If a mutation will cause serious disease, people have this mutation are more likely to die so the freq in database will be low.
3. Familiarize yourself with the hematopoietic hierarchy – what kind of cells can a hematopoietic stem cell generate?
- Hematopoietic Stem Cells: top
- Common Myeloid Progenitors: blood cell
- Common Lymphoid Progenitors: B, T
4. From the paper above, describe other forms of clonal hematopoiesis that are not CHIP.
- Chromosomal alterations
- Loss of chromosomal
- cancer
5. What are some ways that CHIP mutations affect hematopoietic cell phenotype?
- DNMT3A/TET2/ASXL1: increase hematopoietic cell renew
- JAK2 V617F:thrombus
- 
6. Based on the answer to number 5 (and the Cell Reports paper I sent you before), how do you think CHIP can affect cancer cells in other tissues?
- Under chemotherapy, the CHIP gene is more likely to expand (renew advantage)
- Worse overall survival (OS); Higher risk of relapse; Worse progression-free survival (PFS)

## Oct

### 2
- Method:1)Firth logistic regression. 2)PCA. 3)Proposional Harvard Model
- 330 solid tumor gene
### 8
- filter.py: filterd artifacts
- output_final: with flags to germline 
### 9
- filter.py: filterd artifacts
- output_final: with flags to indicate artifacts, germline 
### 14
- CHIP:128, non_CHIP:12989
### 16
- candidate_filter.R: copied from the blood paper
- /output: raw CHIP cohort only contain variant strong associated with age and TERT
### 22
- filter.py: filterd artifacts
- output_final: all_summary.csv: filtered summary with flags
### 24
- filter.py: filterd artifacts, DP<10, minAD<3, AF<2%, binomial test p<0.01, false negative recall if strong associated with age and TERT
- output_final: all_summary.csv: filtered summary with flags
## Nov

### 6
- filter.py: filterd artifacts, DP<10, minAD<3, AF<2%, binomial test p<0.01, false negative recall if strong associated with age and TERT
- output_final: /CHIP: CHIP positive patients; /non_CHIP: CHIP negative patients; all_summary.csv/CHIP)summary.csv/non_CHIP_summary.csv: filtering details
### 13
- CHIP_type.py: script to read output_final to determine CHIP types
- variants_count.py: summarize variants distribution across CHIP cohort
- CHIP_type.csv: CHIP cohort with CHIP types
- CHIP_Variants_summary.csv

### 17
- disease_summary.csv: align phenotype to CHIP status
- summary.csv: Participant ID, CHIP_status, CHIP_type, Cancer_type, Disease_type
### 20
- cohort_vis.R-->cohort_pie.png: CHIP distribution in the cohort
- CHIP_gene_distribution.R-->CHIP_type_summary.png: CHIP types distribution in CHIP positive patients
- cancer_distribution.R-->DiseaseType_vs_CHIP.png: distirbution of cancer type across the whole cohort and the CHIP positive cohort
- CHIP_10.py: encode participant vs. CHIP gene into 1/0 matrix
- matrix_CHIP_binary.csv: row--paricipanf; column--CHIP gene

### 25
- pipeline: candidate_filter.R -> filter.py -> CHIP_type.py -> generate_summary.py ->variants_counts.py -> find_cov.py -> (cohort_vis, cancer_distribution.R, CHIP_gene_distribution.R) -> merged.R -> cohort_matrix.R -> ..
